src/libguac/protocol.c

```diff
@@ -74,6 +75,141 @@ ssize_t __guac_socket_write_length_double(guac_socket* socket, double d) {
 
 }
 
+/**
+ * Converts the image data on a Cairo surface to a JPEG image and sends it
+ * as a base64-encoded transmission on the socket.
+ * 
+ * @param socket
+ *     The guac_socket connection to use.
+ * 
+ * @param surface
+ *     A cairo surface containing the image data to send. 
+ * 
+ * @param quality
+ *     JPEG image quality.
+ * 
+ * @return
+ *     Zero on success, non-zero on error.
+ */
+static int __guac_socket_write_length_jpeg(guac_socket* socket, 
+        cairo_surface_t* surface, int quality) {
+
+    /* Get image surface properties and data */
+    cairo_format_t format = cairo_image_surface_get_format(surface);
+
+    if (format != CAIRO_FORMAT_RGB24) {
+        guac_error = GUAC_STATUS_INTERNAL_ERROR;
+        guac_error_message = "Invalid Cairo image format. Unable to create JPEG.";
+        return -1;
+    }
+
+    int width = cairo_image_surface_get_width(surface);
+    int height = cairo_image_surface_get_height(surface);
+    int stride = cairo_image_surface_get_stride(surface);
+    unsigned char* data = cairo_image_surface_get_data(surface);
+
+    /* Flush pending operations to surface */
+    cairo_surface_flush(surface);
+
+    /* Prepare JPEG bits */
+    struct jpeg_compress_struct cinfo;
+    struct jpeg_error_mgr jerr;
+    cinfo.err = jpeg_std_error(&jerr);
+    jpeg_create_compress(&cinfo);
+
+    unsigned char *mem = NULL;
+    unsigned long mem_size = 0;
```

Comment:
> Yes, it should be set properly so you can remove the initialization.

Ah, excellent. Will do.

> I usually like to initialize variables regardless.

Yeah, there are definitely two warring camps with respect to initialization, and I'm afraid I'm firmly in the don't-do-it-unless-you-actually-need-the-value camp.

I and my fellow campers are irked when initialization would effectively bypass compiler warnings about uninitialized values. Honestly, if I had my way, compilers would also warn if you *assign* a value that is provably never used. ;)

I would argue that, similarly to how writting `assert(0)` indicates semantically to the compiler (or static analyzer) that we assume that case will "never happen", assigning only values that are actually used indicates to the compiler, etc. how we assume the variable will be used, and allows it to warn us if it sees our assumptions are wrong.